packages/pyright-internal/src/localization/package.nls.ru.json

```
@@ -646,7 +646,28 @@
         "yieldOutsideFunction": "\"yield\" не допускается за пределами функции или лямбда-выражении",
         "yieldWithinComprehension": "\"yield\" не допускается внутри включения",
         "zeroCaseStatementsFound": "Оператор match должен включать по крайней мере один оператор case",
-        "zeroLengthTupleNotAllowed": "tuple нулевой длины не допускается в этом контексте"
+        "zeroLengthTupleNotAllowed": "Пустой кортеж не допускается в этом контексте",
+        "argTypeAny": "Тип аргумента — Any",
+        "baseClassAny": "Тип класса неизвестен, потому что базовый класс имеет тип Any",
+        "classDecoratorTypeAny": "Тип класса неизвестен, потому что декоратор имеет тип Any",
+        "classImplicitlyAbstract": "Класс \"{type}\" неявно абстрактный, потому что наследован от абстрактного класса, но не заполнил все абстрактные символы. Если это сделано намеренно, добавьте `ABC` к его родительским классам или используйте `metaclass=ABCMeta`.",
+        "classImplicitlyProtocol": "Класс \"{type}\" неявно класс-протокол, потом что наследован от класса-протокола, но не заполнил все абстрактные символы. Если это сделано намеренно, добавьте `Protocol` или `ABC` к его родительским классам или используйте `metaclass=ABCMeta`.",
+        "functionDecoratorTypeAny": "Тип функции неизвестен, потому что декоратор имеет тип Any",
+        "lambdaReturnTypeAny": "Лямбда-функция возвращает Any",
+        "noneNotUsableWithAsync": "\"None\" нельзя использовать в блоке \"async with\"",
+        "paramTypeAny": "Тип параметра \"{paramName}\" — Any",
+        "positionArgAfterUnpackedDictArg":  "Позиционный аргумент не может следовать за распаковкой именованных аргуметов",
+        "returnTypeAny": "Возвращаемый тип — Any",
+        "typeNotUsableWithAsync": "Объект типа \"{type}\" нельзя использовать в блоке \"async with\", потому что он не имеет корректного метода {method}",
+        "typeAny": "Тип \"{name}\" — Any",
+        "wildcardPatternTypeAny": "Произвольный шаблон захватывает значение типа Any",
+        "pyrightIgnoreCommentWithoutRule": "Комментарий `pyright: ignore` обязан указывать правило (например `# pyright: ignore[baseClassFinal]`)",
```

`baseClassFinal` isn't a rule name it's just a code used for localization and those can't be used in ignore comments. there can be multiple localization codes for a diagnostic rule. in this case `baseClassFinal` is reported under `reportGeneralTypeIssues` so that's the code you'd use to suppress the error
```suggestion
        "pyrightIgnoreCommentWithoutRule": "Комментарий `pyright: ignore` обязан указывать правило (например `# pyright: ignore[reportGeneralTypeIssues]`)",
```

---

My bad, fixed this: https://github.com/DetachHead/basedpyright/pull/1019/commits/acdc4fc6d9519fd4e29c7b88cd5e25d1a4656685

The original rule string says
```
`pyright: ignore` comment must specify a rule (eg. `# pyright: ignore[ruleName]`)
```
which suggests that `# pyright: ignore[ruleName]` is a concrete comment example. Maybe this should also be changed? Or maybe the localization should just translate the placeholder? ( 
```
(например `# pyright: ignore[имя правила]`)
```
)

---

yeah i used `ruleName` not to refer to a specific rule but rather "insert a rule name here". it's probably better to use a specific example though to minimize potential confusion or misinterpretations during localization